File: slack-poster/src/test/java/dev/number6/slackposter/SlackPosterRDG.java
package dev.number6.slackposter;

import dev.number6.slackposter.generate.DoubleAttributeValueGenerator;
import dev.number6.slackposter.generate.IntegerAttributeValueGenerator;
import dev.number6.slackposter.generate.MapAttributeValueGenerator;
import com.amazonaws.services.dynamodbv2.model.AttributeValue;
import uk.org.fyodor.generators.Generator;
import uk.org.fyodor.generators.RDG;
import uk.org.fyodor.range.Range;

public class SlackPosterRDG extends RDG {

    public static Generator<AttributeValue> mapAttributeValues(Generator<String> key, Generator<AttributeValue> val, int size) {
        return new MapAttributeValueGenerator(key, val, size);
    }

    public static Generator<AttributeValue> doubleAttributeValues(double max) {
        return new DoubleAttributeValueGenerator(max);
    }

    public static Generator<AttributeValue> doubleAttributeValues(Range<Double> range) {
        return new DoubleAttributeValueGenerator(range);
    }

    public static Generator<AttributeValue> integerAttributeValues(int max) {
        return new IntegerAttributeValueGenerator(max);
    }

    public static Generator<AttributeValue> integerAttributeValues(Range<Integer> range) {
        return new IntegerAttributeValueGenerator(range);
    }

}

File: slack-poster/src/test/kotlin/dev/number6/slackposter/SlackPosterRDG.kt
package dev.number6.slackposter

import com.amazonaws.services.dynamodbv2.model.AttributeValue
import dev.number6.slackposter.generate.DoubleAttributeValueGenerator
import dev.number6.slackposter.generate.IntegerAttributeValueGenerator
import dev.number6.slackposter.generate.MapAttributeValueGenerator
import uk.org.fyodor.generators.Generator
import uk.org.fyodor.generators.RDG
import uk.org.fyodor.range.Range

object SlackPosterRDG : RDG() {
    fun mapAttributeValues(key: Generator<String>, `val`: Generator<AttributeValue>, size: Int): Generator<AttributeValue> {
        return MapAttributeValueGenerator(key, `val`, size)
    }

    fun doubleAttributeValues(max: Double): Generator<AttributeValue> {
        return DoubleAttributeValueGenerator(max)
    }

    fun integerAttributeValues(max: Int): Generator<AttributeValue> {
        return IntegerAttributeValueGenerator(max)
    }
}